.github/workflows/generator-unit-test.yaml

```diff
@@ -0,0 +1,32 @@
+name: Generator Unit Test
+on:
+  workflow_call:
+  push:
+    branches: [main]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-generator-unit-test
+  cancel-in-progress: true
+jobs:
+  generator-unit-test:
+    name: Generator Unit Test
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain: [latest]
+    steps:
+      - name: Install Swift
+        uses: vapor/swiftly-action@v0.2
+        with:
+          toolchain: ${{ matrix.toolchain }}
+        env:
+          SWIFTLY_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout
+        uses: actions/checkout@v4.2.2
+      - name: Enter the Generator Directory
+        run: cd Generator
```

Comment:
This `cd` doesn't carry over to the next step, meaning we effectively run the root-level unit tests again. You can use the `working-directory` attribute to change the steps into the `Generator` directory. Here's a nice blog post on this: https://cicube.io/blog/github-actions-working-directory

Reply:
Ohh gotcha. Thanks for that insight - I'm much more familiar with GitLab CI! This was adjusted in https://github.com/swift-otel/swift-otel-semantic-conventions/pull/19/commits/65f683870323c144d41edf6c77857a1e709f8964, and the right test set was run here: https://github.com/swift-otel/swift-otel-semantic-conventions/actions/runs/15261721067/job/42920642640?pr=19#step:5:951

Reply:
Yeah, it can definitely be a bit counterintuitive sometimes.